### PR TITLE
Research DB composite types

### PR DESCRIPTION
**Pull request by lbarcziova**
Fixes https://github.com/packit/packit-service/issues/1985

### CODE REVIEW

File: database-composite-types/README.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+# Usage of composite type to store the pipelines
+
+[Composite column types](https://docs.sqlalchemy.org/en/20/orm/composites.html):
```
Comment (jpopelka) [inline database-composite-types/README.md:3]: The link is to SQLAlchemy 2.0, while we're currently using 1.4 (and if we move our images from Fedora to CentOS Stream 9 we'll stay with 1.4 for quite some time), so I'd use the 1.4 version of the docs. 

```suggestion
[Composite column types](https://docs.sqlalchemy.org/en/14/orm/composites.html):
```

Does that affect your examples?

File: database-composite-types/README.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+## 2.option
+
+How could the model look like when using composite type on group level:
+
+- "groups" stored together in one pipeline
+- mapping build -> TF needed
+
+### Issues
```
Comment (jpopelka) [inline database-composite-types/README.md:160]: I see only issues, are there any pros as well? :wink:

Comment (lbarcziova) [reply database-composite-types/README.md:162]: sorry, I thought I included it in the beginning (added now); as the issue suggests, the idea was to make the database design pipeline-centric and remove the need to do the joins on multiple tables

File: database-composite-types/README.md (hunk 1 above)
Comment (jpopelka) [reply database-composite-types/README.md:3]: You marked this as resolved without touching the link or commenting, so unresolving :stuck_out_tongue_winking_eye:

File: database-composite-types/README.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+Examples of when this happens:
+
+- when updating Copr build status in DB, we get the corresponding builds via the build ID from Copr (index)
+- when updating TF run status in DB, we get the corresponding test run via the TF pipeline ID (index)
+- babysit tasks - they get all the pending Copr builds / Test runs
+- when triggering `/packit test` - we get the latest Copr build model with corresponding commit SHA
+
+#### 2. data duplication
+
+- updating SRPM build would require getting all pipelines and updating the data everywhere
```
Comment (majamassarini) [inline database-composite-types/README.md:122]: I agree that data duplication is not a good thing.

**Review by majamassarini — APPROVED** [review]

**Pull request merged**